src/sync/impl/sync_file.cpp

```diff
@@ -248,31 +250,63 @@ std::string SyncFileManager::get_base_sync_directory() const
     return sync_path;
 }
 
-std::string SyncFileManager::user_directory(const std::string& user_identity) const
+std::string SyncFileManager::user_directory(const std::string& local_identity,
+                                            util::Optional<SyncUserIdentifier> user_info) const
 {
-    REALM_ASSERT(user_identity.length() > 0);
-    if (filename_is_reserved(user_identity)) {
+    REALM_ASSERT(local_identity.length() > 0);
+    std::string escaped = util::make_percent_encoded_string(local_identity);
+    if (filename_is_reserved(escaped))
         throw std::invalid_argument("A user can't have an identifier reserved by the filesystem.");
-    }
+
     auto user_path = file_path_by_appending_component(get_base_sync_directory(),
-                                                      user_identity,
+                                                      escaped,
                                                       util::FilePathType::Directory);
-    util::try_make_dir(user_path);
+    bool dir_created = util::try_make_dir(user_path);
+    if (dir_created && user_info) {
+        // Add a text file in the user directory containing the user identity, for backup purposes.
+        // Only do this the first time the directory is created.
+        auto info_path = util::file_path_by_appending_component(user_path, c_user_info_file);
+        std::ofstream info_file;
+        info_file.open(info_path.c_str());
+        if (info_file.is_open()) {
+            info_file << user_info->user_id << "\n" << user_info->auth_server_url << "\n";
+            info_file.close();
+        }
+    }
     return user_path;
 }
 
-void SyncFileManager::remove_user_directory(const std::string& user_identity) const
+void SyncFileManager::remove_user_directory(const std::string& local_identity) const
 {
-    REALM_ASSERT(user_identity.length() > 0);
-    if (filename_is_reserved(user_identity)) {
+    REALM_ASSERT(local_identity.length() > 0);
+    if (filename_is_reserved(local_identity))
```

Comment:
In `user_directory` we percent-encode the identity before checking if the filename is reserved. Here we do it after. This seems like something we should be consistent about.